neuralop/models/gino.py

```diff
@@ -319,51 +345,8 @@ def latent_embedding(self, in_p, ada_in=None):
             in_p = self.fno_blocks(in_p, idx)
 
         return in_p 
-
-    def integrate_latent(self, in_p, out_p, latent_embed):
-        batch_size = latent_embed.shape[0]
-
-        # output shape: (batch, n_out, out_channels) or (n_out, out_channels)
-        in_to_out_nb = self.nb_search_out(
-            in_p.view(-1, in_p.shape[-1]), 
-            out_p,
-            self.gno_radius,
-            )
-    
-        #Embed input points
-        n_in = in_p.view(-1, in_p.shape[-1]).shape[0]
-        if self.pos_embed is not None:
-            in_p_embed = self.pos_embed(in_p.reshape((n_in, -1)))
-        else:
-            in_p_embed = in_p.reshape((n_in, -1))
-        
-        #Embed output points
-        n_out = out_p.shape[0]
-        if self.pos_embed is not None:
-            out_p_embed = self.pos_embed(out_p.reshape((n_out, -1)))
-        else:
-            out_p_embed = out_p #.reshape((n_out, -1))
-                
-        #latent_embed shape b, c, n_1, n_2, ..., n_k
-        latent_embed = latent_embed.permute(0, *self.in_coord_dim_reverse_order, 1).reshape(batch_size, -1, self.fno_hidden_channels)
-        # shape b, n_out, channels
-        if self.out_gno_tanh in ['latent_embed', 'both']:
-            latent_embed = torch.tanh(latent_embed)
-        
-        #(n_out, fno_hidden_channels)
-        out = self.gno_out(y=in_p_embed, 
-                    neighbors=in_to_out_nb,
-                    x=out_p_embed,
-                    f_y=latent_embed,)
-                    #weighting_fn=self.gno_weighting_fn)
-        out = out.permute(0, 2, 1)
-        # Project pointwise to out channels
-        #(b, n_in, out_channels)
-        out = self.projection(out).permute(0, 2, 1)  
-
-        return out
     
-    def forward(self,  x, input_geom, latent_queries, output_queries, ada_in=None, **kwargs):
+    def forward(self,  x, input_geom, latent_queries, output_queries, latent_features=None, ada_in=None, **kwargs):
         """forward pass of GNO --> latent embedding w/FNO --> GNO out
```

Comment:
I wonder what is the meaning of "latent embedding w/FNO" here?

Reply:
Cleaned up a bit, thanks